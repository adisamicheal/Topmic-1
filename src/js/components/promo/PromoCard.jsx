// ########## Import Dependencies Here ##########
import React from 'react';
import { shape, number, string } from 'prop-types';

// ########## Import Screens Here ##########

// ########## Import Components Here ##########
import ReadMoreButton from './ReadMoreButton';

const PromoCard = (props) => {
  const { data } = props;
  return (
    <div className="promo-card">
      <div className="wrapper">
        <div className={`promo-card-container-${data.promoCardId > 2 && "sectionTwo-" || data.promoCardId <= 2 && "sectionOne-"}${data.imageDirection}`}>
          <div className={`promo-image-container-${data.promoCardId > 2 && "sectionTwo-" || data.promoCardId <= 2 && "sectionOne-"}${data.imageDirection}`}>
            <div className={`promo-image-${data.promoCardId > 2 && "sectionTwo-" || data.promoCardId <= 2 && "sectionOne-"}${data.imageDirection}`} />
          </div>
          <div className="promo-text-container">
            <div className="promo-text-wrapper">
              <h4 className="promo-heading">{data.title}</h4>
              <p className="promo-description">{data.description}</p>
              <ReadMoreButton
                align="left"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

PromoCard.propTypes = {
  data: shape({
    promoCardId: number.isRequired,
    imageDirection: string.isRequired,
    title: string.isRequired,
    description: string.isRequired
  }).isRequired
}

export default PromoCard;