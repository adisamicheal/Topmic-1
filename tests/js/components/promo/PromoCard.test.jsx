import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PromoCard from '../../../../src/js/components/promo/PromoCard';

Enzyme.configure({ adapter: new Adapter() });

describe('testing PromoCard component', () => {
  const data = {
    promoCardId: 1,
    title: "Our School",
    description: "We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The schoolâ€™s co-curricular activities and unique programs all reinforce our desire to develop the whole child.",
    imageLink: "./img/our-school-photo.jpg",
    readMoreLink: "",
    imageDirection: "left"
  };
  const wrapper = shallow(<PromoCard data={data} />);
  it('renders the PromoCard component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});