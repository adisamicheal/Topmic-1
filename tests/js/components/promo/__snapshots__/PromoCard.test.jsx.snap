// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing PromoCard component renders the PromoCard component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PromoCard
    data={
      Object {
        "description": "We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.",
        "imageDirection": "left",
        "imageLink": "./img/our-school-photo.jpg",
        "promoCardId": 1,
        "readMoreLink": "",
        "title": "Our School",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="wrapper"
      >
        <div
          className="promo-card-container-sectionOne-left"
        >
          <div
            className="promo-image-container-sectionOne-left"
          >
            <div
              className="promo-image-sectionOne-left"
            />
          </div>
          <div
            className="promo-text-container"
          >
            <div
              className="promo-text-wrapper"
            >
              <h4
                className="promo-heading"
              >
                Our School
              </h4>
              <p
                className="promo-description"
              >
                We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
              </p>
              <ReadMoreButton
                align="left"
              />
            </div>
          </div>
        </div>
      </div>,
      "className": "promo-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="promo-card-container-sectionOne-left"
        >
          <div
            className="promo-image-container-sectionOne-left"
          >
            <div
              className="promo-image-sectionOne-left"
            />
          </div>
          <div
            className="promo-text-container"
          >
            <div
              className="promo-text-wrapper"
            >
              <h4
                className="promo-heading"
              >
                Our School
              </h4>
              <p
                className="promo-description"
              >
                We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
              </p>
              <ReadMoreButton
                align="left"
              />
            </div>
          </div>
        </div>,
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="promo-image-container-sectionOne-left"
            >
              <div
                className="promo-image-sectionOne-left"
              />
            </div>,
            <div
              className="promo-text-container"
            >
              <div
                className="promo-text-wrapper"
              >
                <h4
                  className="promo-heading"
                >
                  Our School
                </h4>
                <p
                  className="promo-description"
                >
                  We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                </p>
                <ReadMoreButton
                  align="left"
                />
              </div>
            </div>,
          ],
          "className": "promo-card-container-sectionOne-left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="promo-image-sectionOne-left"
              />,
              "className": "promo-image-container-sectionOne-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "promo-image-sectionOne-left",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="promo-text-wrapper"
              >
                <h4
                  className="promo-heading"
                >
                  Our School
                </h4>
                <p
                  className="promo-description"
                >
                  We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                </p>
                <ReadMoreButton
                  align="left"
                />
              </div>,
              "className": "promo-text-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4
                    className="promo-heading"
                  >
                    Our School
                  </h4>,
                  <p
                    className="promo-description"
                  >
                    We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                  </p>,
                  <ReadMoreButton
                    align="left"
                  />,
                ],
                "className": "promo-text-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Our School",
                    "className": "promo-heading",
                  },
                  "ref": null,
                  "rendered": "Our School",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.",
                    "className": "promo-description",
                  },
                  "ref": null,
                  "rendered": "We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "align": "left",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="wrapper"
        >
          <div
            className="promo-card-container-sectionOne-left"
          >
            <div
              className="promo-image-container-sectionOne-left"
            >
              <div
                className="promo-image-sectionOne-left"
              />
            </div>
            <div
              className="promo-text-container"
            >
              <div
                className="promo-text-wrapper"
              >
                <h4
                  className="promo-heading"
                >
                  Our School
                </h4>
                <p
                  className="promo-description"
                >
                  We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                </p>
                <ReadMoreButton
                  align="left"
                />
              </div>
            </div>
          </div>
        </div>,
        "className": "promo-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="promo-card-container-sectionOne-left"
          >
            <div
              className="promo-image-container-sectionOne-left"
            >
              <div
                className="promo-image-sectionOne-left"
              />
            </div>
            <div
              className="promo-text-container"
            >
              <div
                className="promo-text-wrapper"
              >
                <h4
                  className="promo-heading"
                >
                  Our School
                </h4>
                <p
                  className="promo-description"
                >
                  We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                </p>
                <ReadMoreButton
                  align="left"
                />
              </div>
            </div>
          </div>,
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="promo-image-container-sectionOne-left"
              >
                <div
                  className="promo-image-sectionOne-left"
                />
              </div>,
              <div
                className="promo-text-container"
              >
                <div
                  className="promo-text-wrapper"
                >
                  <h4
                    className="promo-heading"
                  >
                    Our School
                  </h4>
                  <p
                    className="promo-description"
                  >
                    We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                  </p>
                  <ReadMoreButton
                    align="left"
                  />
                </div>
              </div>,
            ],
            "className": "promo-card-container-sectionOne-left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="promo-image-sectionOne-left"
                />,
                "className": "promo-image-container-sectionOne-left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "promo-image-sectionOne-left",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="promo-text-wrapper"
                >
                  <h4
                    className="promo-heading"
                  >
                    Our School
                  </h4>
                  <p
                    className="promo-description"
                  >
                    We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                  </p>
                  <ReadMoreButton
                    align="left"
                  />
                </div>,
                "className": "promo-text-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="promo-heading"
                    >
                      Our School
                    </h4>,
                    <p
                      className="promo-description"
                    >
                      We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.
                    </p>,
                    <ReadMoreButton
                      align="left"
                    />,
                  ],
                  "className": "promo-text-wrapper",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Our School",
                      "className": "promo-heading",
                    },
                    "ref": null,
                    "rendered": "Our School",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.",
                      "className": "promo-description",
                    },
                    "ref": null,
                    "rendered": "We would like to extend a warm welcome to Winsome School System and take a few moments to share our vision with you. We are a healthy, thriving educational community catering to students in Childcare through to Year 10.The school has enjoyed a strong academic achievement record, and an engaging, varied curriculum. The school’s co-curricular activities and unique programs all reinforce our desire to develop the whole child.",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "align": "left",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
